rir/src/ir/BC_inc.h

```diff
@@ -140,33 +140,16 @@ class BC {
 
     BC() : bc(Opcode::invalid_) {}
 
-    BC(Opcode* pc) {
-        bc = *pc;
-        pc++;
-        immediate = decodeImmediateArguments(bc, pc);
-        // Read implicit promise argument offsets
-        if (bc == Opcode::call_implicit_ ||
-            bc == Opcode::named_call_implicit_) {
-            pc += sizeof(CallFixedArgs);
-            for (size_t i = 0; i < immediate.callFixedArgs.nargs; ++i)
-                immediateCallArguments.push_back(readImmediate(&pc));
-        }
-        // Read named arguments
-        if (bc == Opcode::named_call_ || bc == Opcode::named_call_implicit_) {
-            for (size_t i = 0; i < immediate.callFixedArgs.nargs; ++i)
-                callArgumentNames.push_back(readImmediate(&pc));
-        }
-    }
-
-    BC operator=(BC other) {
-        bc = other.bc;
-        immediate = other.immediate;
-        return other;
-    }
+    BC& operator=(BC other) = delete;
+    BC(const BC& other) = delete;
+    BC(const BC&& other)
```

Comment:
1/ shouldn't it be `BC(BC&& other)` for move ctor?
2/ move assignment is missing

Reply:
hmm, good point. I wonder why it even works